nixos/modules/services/matrix/synapse.nix

```diff
@@ -195,15 +202,16 @@ in {
     listenerType = workerContext: types.submodule {
```

Comment:
```suggestion
    listenerType = workerContext: types.submodule ({config, ...}: {
```

Reply:
I fixed the filtering instead.

Reply:
This is still required to set the correct defaults. Without this we can't set defaults based on `config.path`.
You can use `nix-build -A nixosTests.matrix-synapse.nodes.serverpostgres.system.build.toplevel; grep -o '/nix/store/.*-homeserver\.yaml' result/etc/systemd/system/matrix-synapse.service | xargs cat` to see that the default listener for synapse still contains bind addresses, port *and* mode despite path being null.

Reply:
Yeah, I noticed. Will be in the next update.